test/integration/integration_test.py

```diff
@@ -0,0 +1,144 @@
+## The integration tests consists of a back and forth between python (which we use to create and validate
+## image queries) and bash (which we use to check deployments are properly rolled out)
+## This file contains all the modes that we use for integeration testing.
+## Modes:
+## - Create the integration test detector
+## - Submit the initial dog/cat image query to the edge, expect low confidence
+## - Train the edge model by submitting image queries to the cloud.
+## - Submit the final dog/cat image query to the edge, expect high confidence
+
+import argparse
+import random
+import time
+
+from groundlight import Groundlight, GroundlightClientError
+from model import Detector
+
+NUM_IQS_TO_IMPROVE_MODEL = 10
+ACCETABLE_TRAINED_CONFIDENCE = 0.8
+
+
+def get_groundlight():
+    try:
+        return Groundlight(endpoint="http://localhost:30107")
+    except GroundlightClientError:
+        # we use this to create a detector since we do that before setting up edge
+        # although maybe we want to be more careful here about making sure that's
+        # the case we're in
+        return Groundlight()
+
+
+gl = get_groundlight()
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Submit a dog and cat image to k3s Groundlight edge-endpoint for integration tests"
+    )
+    parser.add_argument(
+        "-m",
+        "--mode",
+        type=str,
+        choices=["create_detector", "initial", "improve_model", "final"],
+        help="Mode of operation: 'initial', 'many', or 'final'",
+        required=True,
+    )
+    parser.add_argument("-d", "--detector_id", type=str, help="id of detector to use", required=False)
+    args = parser.parse_args()
+
+    detector = None
+    if args.detector_id:
+        detector = gl.get_detector(args.detector_id)
+
+    if detector is None and args.mode != "create_detector":
+        raise ValueError("You must provide detector id unless mode is create detector")
+
+    if args.mode == "create_detector":
+        detector_id = create_cat_detector()
+        print(detector_id)  # print so that the shell script can save the value
+    elif args.mode == "initial":
+        submit_initial(detector)
+    elif args.mode == "improve_model":
+        improve_model(detector)
+    elif args.mode == "final":
+        submit_final(detector)
+
+
+def create_cat_detector() -> str:
+    """Create the intial cat detector that we use for the integration tests. We create
+    a new one each time."""
+    random_number = random.randint(0, 9999)
```

Comment:
Is it worth increasing the range here just to make it even more unlikely that we get a collision? You could also generate a ksuid like we do [here](https://github.com/groundlight/edge-endpoint/blob/2db00f7568469ac0d09b357ae349f2a0c6ee65b0/app/core/utils.py#L157) to ensure there's no issues.